### ISSUE TITLE
pandoc conversion to odt with images / missing alt-Text

### ISSUE THREAD

**Issue by slartibart70** [jgm/pandoc#6354]
I'm using pandoc-markdown to convert md to odt (libreOffice writer) and i want to see the caption/alt-text in the resulting odt document. This works fine, as long as an image is not contained in a list, then the alt-text gets suppressed.
Example:
```
# some heading

- bulleted text
  
  bulleted text continuation
    
  ![alt-text not being displayed](images/zip-1.png){ width=60% }

Normal Text
continuation:

![alt-text is visible](images/zip-1.png){ width=60% }
```

If you generate a html output, you get (in both cases above) sometihng like

```
<ul>
<li>
<figure>
<img src="images/img1.png" style="width:60.0%" alt="" /><figcaption>alternative-text or caption-text</figcaption>
</figure></li>
</ul>
```

But, in the ODT output this gets lost

```
<text:list text:style-name="L1">
  <text:list-item>
    <text:p text:style-name="P1">bulleted text</text:p><text:p text:style-name="P1">bulleted text continuation</text:p><text:p text:style-name="P1"><draw:frame draw:name="img1" style:rel-width="60.0%" style:rel-height="scale" svg:width="516.0875912408759pt" svg:height="422.014598540146pt"><draw:image xlink:href="Pictures/0.png" xlink:type="simple" xlink:show="embed" xlink:actuate="onLoad" /></draw:frame></text:p>
  </text:list-item>
</text:list>
<text:p text:style-name="First_20_paragraph">Normal Text continuation:</text:p>
<text:p text:style-name="FigureWithCaption"><draw:frame draw:name="img2" style:rel-width="60.0%" style:rel-height="scale" svg:width="516.0875912408759pt" svg:height="422.014598540146pt"><draw:image xlink:href="Pictures/1.png" xlink:type="simple" xlink:show="embed" xlink:actuate="onLoad" /></draw:frame></text:p>
<text:p text:style-name="FigureCaption">alt-text is visible</text:p>
```

So, the whole FigureCaption is not there any more if it's contained in a text-list.
Any idea how to fix this?

And, as long as we are on this ODT topic,
how can we achieve that an image included into an ODT is not rendered as a character inside a paragraph but sth. else (like being an independent image having an anchor to the paragraph)?
To see this behaviour, you need to open the odt once in LibreOffice and save it again, then you get  the following:

```
<draw:frame draw:style-name="fr1" draw:name="img1" text:anchor-type="as-char" ...
````

**Comment by mistmist** [comment]
cannot reproduce this; was fixed presumably by commit 909ced5153e2c7cefd5018c39f83231824940fb8 "Support complex figures. [API change]"

the parent of this commit reproduces the bug, the next 6 or so commits don't build with stack and the next one that builds doesn't reproduce the bug.

regarding the positioning question, ODF has 5 different anchor types where "paragraph" or "char" would be suitable for pandoc; it requires a lot of additional attributes like svg:x, svg:y, style:vertical-pos, style:vertical-rel, style:horizontal-pos, style:horizontal-rel, style:wrap, some of which i guess may need to be translated from the import format.

in any case that should be reported as a separate issue/feature request.

hmm, apparently Docx.hs only produces wp:inline (same as text:anchor-type="as-char") so there isn't any inspiration to draw from there.

**Comment by jgm** [comment]
I'll close this then as fixed. If there are other things remaining, please open new issues for them.
